.github/workflows/cd.yml

```diff
@@ -11,6 +11,10 @@ env:
   # used to rename and upload the binaries
   PROJ_NAME: fj-app
 
+  # This lets our app know it's an "official" release. Otherwise we would get
+  # a version number like "fj-app 0.8.0 (8cb928bb, unreleased)"
+  FJ_OFFICIAL_RELEASE: 1
```

Comment:
I believe this is incorrect. The CD workflow is run (and builds binaries for) every push to `main`. If a release is detected, those binaries are re-used for the actual release.

Note the `Operator | Deduce` step, which detects whether we're dealing with a release, or a regular push to `main`, and the `if: ${{ steps.release.outputs.release-detected == 'true' }}` bits that use that step's output.

Not sure how good your grasp on GitHub Actions is. We might need @hendrikmaus to help figure this out.

If this turns into a problem, we could consider removing the per-push-to-`main` build, and only build binaries for the release. I'm not sure how useful those per-push binaries are anymore, with weekly releases. Thoughts?

Reply:
Ahh, I just had a deeper look into how the release job works, and it seems like every push to `main` generates its own compiled binaries, but they only get promoted to a proper release if the "release operator" says so.

That makes things more complicated because `fj-app` won't know at compile time whether it is a proper release (the `release` job depends on `binaries`, but `Binaries | Compile` would need to know `steps.release.outputs.release-detected` - which is a dependency cycle).